### ISSUE TITLE
Icon spacing changes when toggling on Monterey

### ISSUE THREAD

**Issue by kprav33n** [Mortennn/Dozer#169]
I'm using 4.0.0 (29) on macOS 12.0.1 (Monterey). When toggling the expanded icons, the position of icons on the right side of the dots change a lot because of spacing. Is this expected? I would expect that the icons on the right side of the dots remain in the same place before and after expanding the hidden icons. See the following animation.

![DozerResizing](https://user-images.githubusercontent.com/659069/144000521-b033cadf-f931-47d5-b4a0-2016e8d4eed7.gif)

**Comment by johsoderi** [comment]
It seems like Dozer places the icons closer together, and upon expanding it reverts to the macOS original spacing.
I would love to have the tighter spacing also when expanded, or for issue #168 to be resolved. As it is now, the leftmost icons are hidden behind the notch when I expand.

Edit: After some more testing I notice that if I turn on the option "Hide both Dozer icons when status bar icons are hidden" and expand the icons with a shortcut, the spacing doesn't change - most of the time. It's inconsistent though, sometimes it does.

Dozer v. 4.0.0 (29) - macOS Monterey 12.0.1 - Macbook Pro M1 Max 16" - Default screen res. (3456x2234)

**Comment by aonez** [comment]
I'm not getting this issue on Monterey 12.0.1 (21A559). I'm using a 4.2.1 [custom Apple Silicon build](https://github.com/aonez/Dozer/releases/tag/v4.2.1-silicon) though, but tested with 4.0.0 and also no issue.

I suppose since @johsoderi stated M1 Max and the gif menu bar from @kprav33n seems very tall we can assume both of you have a notch and this issue will only be present in that taller menu bar size.

I don't own a notch Mac so sadly can't test this one.

**Comment by aonez** [comment]
Looking more slowly to the gif probably Apple made the space between menu bar items dynamic so it fits more before they reach the notch. Since Dozer uses a very big item to hide the rest of the items, when hiding them the space between items is the minimum and while showing the items it must be the maximum size.

**Comment by Taraan03** [comment]
I'm seeing this issue as well. I'm on a notch Mac (16-inch, 2021), with MacOS monterey 12.0.1 (21A559). Happy to share other info if it would help debug.

EDIT: I also tried @aonez apple silicon build and got the same behavior.

**Comment by aonez** [comment]
@Taraan03 maybe you can try with this simple app to see if the space between bar menu items shrinks when they approach the notch. Close Dozer first, so the items are not hidden.

[NotchMenuBarMarginTest.zip](https://github.com/Mortennn/Dozer/files/7688069/NotchMenuBarMarginTest.zip)

Simply add enough bar items with the sample app so they reach or pass the notch. You can also add an space item for the same test.

**Comment by johsoderi** [comment]
I gave it a try. Not sure what to make out of it though, is this the behaviour Apple intended? Looks buggy to me.
![menubar](https://user-images.githubusercontent.com/31351786/145493288-f30dbcb7-836b-4e6e-afd8-96f7716c8938.gif)

**Comment by aonez** [comment]
Hm, so it snaps to the notch using dynamic margins when it's near it, not before. What happens when the icons arrive at the notch? They appear in the left of it or do they simply disappear?

I think a simple workaround can be done using proper size spacers.

**Comment by johsoderi** [comment]
They just disappear, nothing shows to the left of the notch no matter how many icons I add.

**Comment by kprav33n** [comment]
Looks like macOS 12.1 has some fixes for the menu bar problems with notch. It should be out next week. I'll test with that and update here.

**Comment by unphased** [comment]
Pretty sure the change in display regime of icons from right-aligned to justified is built-in macOS behavior. Issues like this abound in all menu bar apps on github!

**Comment by blakegearin** [comment]
> Looks like macOS 12.1 has some fixes for the menu bar problems with notch. It should be out next week. I'll test with that and update here.

`12.1` hasn't changed this behavior for me.

**Comment by unphased** [comment]
Agreed, what is the nature of the fix that 12.1 was supposed to introduce?
